server/index.js

```diff
@@ -19,72 +19,253 @@ app.use(express.static('./react-client/dist'));
+    return res.status(400).send('No itemImages present in request query params');
+  }
+
+  Images.fetchItemImages(itemId)
+    .then((data) => {
+      if (!data) {
+        res.status(400).send(`Item with the itemId ${itemId} does not exist`);
+      } else {
+        const potentialErrorMessage = determineValidItemData(itemId, itemImages);
+
+        if (potentialErrorMessage) {
+          res.status(400).send(potentialErrorMessage);
+        } else {
+          Images.updateRecord({ itemId, itemImages })
+            .then((data) => {
+              res.status(201).send(`Item ${itemId} succesfully updated`);
+            })
+            .catch((err) => {
+              console.log(err);
+              res.status(500).send(err);
+            });
+        }
+      }
     })
-  } else {
-    Images.fetchItemImages(itemId)
+    .catch((err) => {
+      console.log(err);
+      res.status(500).send(err);
+    });
+});
+
+app.delete('/deleteItemImages/:itemId', (req, res) => {
+  const { itemId } = req.params;
+
+  if (itemId) {
+    const parsedItemId = Number.parseInt(itemId, 10);
+
+    if (parsedItemId < 100 || !Number.isInteger(parsedItemId)) {
+      return res.status(400).send('itemId invalid. Must be string representing integer number greater than 99');
+    }
+  }
+
+  Images.fetchItemImages(itemId)
     .then((data) => {
       if (data) {
-        res.status(200).send({image: data.itemImages[0].small});
+        Images.deleteRecord(itemId)
+          .then((data) => {
+            res.status(201).send(`Item ${itemId} deleted`);
+          })
+          .catch((err) => {
+            console.log(err);
+            res.status(500).send(err);
+          });
       } else {
-        res.sendStatus(404);
+        res.status(400).send('No item matches itemId');
       }
     })
     .catch((err) => {
-      res.status(500).send(err);
       console.log(err);
-    })
+      res.status(500).send(err);
+    });
+});
+
+app.get('/itemImages/:itemId/mainImage', function(req, res) {
+  const itemId = req.params.itemId;
+
+  if (itemId.includes('array')) {
+    const itemsInArray = itemId.substring(5);
+    const itemIds = itemsInArray.split(',');
+
+    for (let i = 0; i < itemIds.length; i++) {
+      const itemIdParsed = Number.parseInt(itemIds[i], 10);
+
+      if (itemIdParsed < 100 || itemIdParsed > 199) {
+        res.status(404).send('Item IDs not valid');
+        return;
+      }
+    }
+
+    Images.fetchMultipleItemImages(itemIds)
+      .then((data) => {
+        if (data) {
+          const responseData = [];
+          data.forEach((item) => {
+            const parsedData = {
+              itemId: item.itemId,
+              image: item.itemImages[0].small,
+            };
+
+            responseData.push(parsedData);
+          });
+          res.status(200).send(responseData);
+        } else {
+          res.sendStatus(404);
+        }
+      })
+      .catch((err) => {
+        res.status(500).send(err);
```

Comment:
Hmm 500 code. I need to look into that... It's just a catch all im reading? For when something goes wrong outside of your error handling?

Reply:
Yes